enhancements/network/ingress-node/ingress-node-firewall-with-ebpf.md

```diff
@@ -0,0 +1,446 @@
+---
+title: ingress-node-firewall-with-ebpf
+
+authors:
+  - "@msherif1234"
+
+reviewers:
+  - "@danwinship"
+  - "@dave-tucker"
+  - "@dcbw"
+  - "@trozet"
+
+approvers:
+  - "@danwinship"
+  - "@knobunc"
+
+api-approvers:
+  - "@danwinship"
+
+creation-date: 2022-06-15
+
+last-updated: 2022-06-21
+
+status: implementable
+
+tracking-link:
+  - https://issues.redhat.com/browse/SDN-2781
+---
+
+# Ingress node Firewall using eBPF
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+OCP customers would like to secure OCP's nodes from external attacks by configuring ingress firewall policies.
+eBPF provides a very flexible filtering mechansim to allow early detection and filtering.
+
+New cluster scoped object `IngressNodeFirewall` will be used to configure node ingress firewall match rules and a per matching rule statistics will be avaliable.
+Admin can provision multiple instances of this object while ensuring certain excuetion order.
+Another cluster scoped object `NodeEndpoint` which will be used to specify the attachment interface(s) for example admin can apply specific node firewall rules on one or more interfaces and another different set of rules on another interface(s) or apply the same rules to all interfaces.
+
+## Motivation
+
+Customers would like to secure OCP cluster from external attacks.
+
+### Goals
+
+- flexible match filters rules for customer to use with a CRD validation to detect and fail invalid configuration.
+- Provide a mechanism for customers to select which nodes via `nodeSelector`.
+- Provide a mechanism to define attachement interfaces for the node ingress firewall objects.  
+- Ingress node firewall is deployed on a fully operational OCP cluster (i.e. cluster comes up with no default ingress firewall policy).
```

Comment:
I'm not sure I follow. Do you mean that if you deploy the firewall on a cluster that it will default to being disabled?
Surely enabled but with the fail-safe rules you call out below might be a safer choice

Reply:
the default when the cluster is up is no ingress node fw I will rewrite it to make it more clear